src/eos-metrics-instrumentation.c

```diff
@@ -423,23 +429,44 @@ login_dbus_proxy_new (void)
                                      "org.freedesktop.login1.Manager",
                                      NULL /* GCancellable */,
                                      &error);
-    if (dbus_proxy == NULL)
+    if (error)
       {
-        g_warning ("Error creating GDBusProxy: %s\n", error->message);
+        g_warning ("Error creating GDBusProxy: %s.", error->message);
         g_error_free (error);
+        return NULL;
       }
-    else
+
+    g_signal_connect (dbus_proxy, "g-signal", G_CALLBACK (record_login),
+                      NULL /* data */);
+
+    GVariant *sessions =
+      g_dbus_proxy_call_sync (dbus_proxy, "ListSessions",
```

Comment:
nit: Can squish this call onto fewer lines now that you broke the line after the assignment.